contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -3346,4 +3349,63 @@ remove_delimited_identifer(char *str)
 	len = strlen(str);
 	while (isspace(str[len - 1]))
 		str[--len] = 0;
-}
\ No newline at end of file
+}
+
+Datum
+sp_enum_oledb_providers_internal(PG_FUNCTION_ARGS)
+{	
+	int			rc;
+	MemoryContext savedPortalCxt;
+
+	/* SPI call input */
+	const char *query = "SELECT "
+                    		"CAST('TDS_FDW' AS sys.nvarchar(255)) AS \"Provider Name\", "
+                    		"CAST('{' || uuid_in(md5(('A PostgreSQL foreign data wrapper to TDS databases ' || subquery.extversion)::text)::cstring) || '}' AS sys.nvarchar(255)) AS \"Parse Name\", "
+                    		"CAST('A PostgreSQL foreign data wrapper to connect to TDS databases ' || subquery.extversion AS sys.nvarchar(255)) AS \"Provider Description\" "
```

Comment:
can't we reference the Provider Description column directly and hash it to get Parse Name